### DISCUSSION TITLE
Schema::hasColumn() not working with sqlite3 on freebsd

### DISCUSSION THREAD

**Discussion by garrettboone** [laravel/framework#51562] [Q&A] [answered]
Hi, I am running into a bug during testing which is reported as:
`SQLSTATE[HY000]: General error: 1 no such column: users_sockets (Connection: sqlite, SQL: select name, type, not "notnull" as "nullable", dflt_value as "default", pk as "primary", hidden as "extra" from pragma_table_xinfo("users_sockets") order by cid asc)`

There is a migration step which is causing the fault:
```
 if (! Schema::hasColumn('users_sockets', 'created_at')) {
    Schema::table('users_sockets', function (Blueprint $table) {
        $table->timestamp('created_at');
    });
}
```

This is on a FreeBSD instance, Laravel 11, sqlite3 version 3.45.1,1, php8.3

Is this expected? The migrations work fine when running on a mariadb database. 

Also tests are working fine when run on a different Ubuntu WSL2 instance same Laravel version and sqlite3 version 3.37.2-2ubuntu0.3. (At least they seem to be working alright as there are no errors.)

**Comment by garrettboone** [comment]
I am still unable to determine why this error is occurring. `Schema::getTables()` is working ok during testing with sqlite connection.

I had DB_DATABASE for testing set to `:memory:` but even when changed to local sqlite db, it fails.

```
php artisan db:table users_sockets

  users_sockets ......................................................................................
  Columns .......................................................................................... 4
  Size ...................................................................................... 16.00 KB

  Column ........................................................................................ Type
  user_id int ................................................................................ int(11)
  socket_id varchar, utf8mb4_unicode_ci ................................................. varchar(191)
  created_at timestamp, nullable ...................................................... NULL timestamp
  updated_at timestamp, nullable ...................................................... NULL timestamp
```

**Reply by garrettboone** [reply]
OK, I think I've narrowed it down to there is not actually any table, although the error is saying no column. Testing is not picking up the schema files before the individual migrations.

**Comment by MRGGC** [comment]
Hi @garrettboone, I am facing the exact same issue on FreeBSD. Did you happen to find a solution to this, by chance?..

**Comment by garrettboone** [comment, marked as answer]
@MRGGC Yes I believe it was fixed here after I reported as a bug  https://github.com/laravel/framework/issues/51609
